app/admin/quests/create/page.tsx

```diff
@@ -0,0 +1,1049 @@
+      showNotification("Start time cannot be less than current time", "info");
+      return;
+    }
+
+    setQuestInput((prev) => ({
+      ...prev,
+      start_time: new Date(startTime).getTime(),
+    }));
+  }, [startTime]);
+
+  useEffect(() => {
+    // check if start_time is less than end_time
+    if (new Date(endTime).getTime() < new Date(startTime).getTime()) {
+      showNotification("End time cannot be less than start time", "info");
+      return;
+    }
+    setQuestInput((prev) => ({
+      ...prev,
+      expiry: new Date(endTime).getTime(),
+    }));
+    setBoostInput((prev) => ({
+      ...prev,
+      expiry: new Date(endTime).getTime(),
+    }));
+  }, [endTime]);
+
+  const handlePublishQuest = useCallback(
+    async (value: boolean) => {
+      await AdminService.updateQuest({
+        id: questId,
+        disabled: value,
+      });
+      await fetchQuestData();
+    },
+    [questId]
+  );
+
+  const handleCreateTask = useCallback(async () => {
+    steps.map(async (step) => {
+      if (step.type === "Quiz") {
+        const response = await AdminService.createQuiz({
+          quest_id: questId,
+          name: step.data.quiz_name,
+          desc: step.data.quiz_desc,
+          intro: step.data.quiz_intro,
+          cta: step.data.quiz_cta,
+          help_link: step.data.quiz_help_link,
+        });
+        await Promise.all(
+          step.data.questions.map(
+            async (question: typeof QuizQuestionDefaultInput) => {
+              await AdminService.createQuizQuestion({
+                quiz_id: response.id,
+                question: question.question,
+                options: question.options,
+                correct_answers: question.correct_answers,
+              });
+            }
+          )
+        );
+      }
+      if (step.type === "TwitterFw") {
+        await AdminService.createTwitterFw({
+          quest_id: questId,
+          name: step.data.twfw_name,
+          desc: step.data.twfw_desc,
+          username: step.data.twfw_username,
+        });
+      } else if (step.type === "TwitterRw") {
+        await AdminService.createTwitterRw({
+          quest_id: questId,
+          name: step.data.twrw_name,
+          desc: step.data.twrw_desc,
+          post_link: step.data.twrw_post_link,
+        });
+      } else if (step.type === "Discord") {
+        await AdminService.createDiscord({
+          quest_id: questId,
+          name: step.data.dc_name,
+          desc: step.data.dc_desc,
+          invite_link: step.data.dc_invite_link,
+          guild_id: step.data.dc_guild_id,
+        });
+      } else if (step.type === "Custom") {
+        await AdminService.createCustom({
+          quest_id: questId,
+          name: step.data.custom_name,
+          desc: step.data.custom_desc,
+          cta: step.data.custom_cta,
+          href: step.data.custom_href,
+        });
+      }
+    });
+    setCurrentPage((prev) => prev + 1);
+  }, [steps]);
+
+  return (
```

Comment:
After reading I feel that this page is way to big and very difficult to maintain and organize. 

I'm pretty sure we can do subcomponents here to make stuff clearer (so it won't be UI component but it'll be what I call a product or business component, they won't probably be reused elsewhere but they permit us to have a very well-organized code). 

Here is a GPT 4 recommendation to refactor https://chatgpt.com/share/805f1ed6-42f0-42ab-aed8-7585243c18a8